source/server/http/admin.cc

```diff
@@ -640,23 +640,32 @@ Http::Code AdminImpl::handlerResetCounters(absl::string_view, Http::HeaderMap&,
   return Http::Code::OK;
 }
 
-Http::Code AdminImpl::handlerServerInfo(absl::string_view, Http::HeaderMap& headers,
-                                        Buffer::Instance& response, AdminStream&) {
-  time_t current_time = time(nullptr);
-  envoy::admin::v2alpha::ServerInfo server_info;
-  server_info.set_version(VersionInfo::version());
+envoy::admin::v2alpha::ServerInfo::State AdminImpl::serverState() {
+  envoy::admin::v2alpha::ServerInfo::State state;
 
   switch (server_.initManager().state()) {
   case Init::Manager::State::Uninitialized:
-    server_info.set_state(envoy::admin::v2alpha::ServerInfo::PRE_INITIALIZING);
+    state = envoy::admin::v2alpha::ServerInfo::PRE_INITIALIZING;
     break;
   case Init::Manager::State::Initializing:
-    server_info.set_state(envoy::admin::v2alpha::ServerInfo::INITIALIZING);
+    state = envoy::admin::v2alpha::ServerInfo::INITIALIZING;
+    break;
+  case Init::Manager::State::Initialized:
+    state = server_.healthCheckFailed() ? envoy::admin::v2alpha::ServerInfo::DRAINING
+                                        : envoy::admin::v2alpha::ServerInfo::LIVE;
     break;
-  default:
-    server_info.set_state(server_.healthCheckFailed() ? envoy::admin::v2alpha::ServerInfo::DRAINING
-                                                      : envoy::admin::v2alpha::ServerInfo::LIVE);
   }
+
+  return state;
+}
+
+Http::Code AdminImpl::handlerServerInfo(absl::string_view, Http::HeaderMap& headers,
+                                        Buffer::Instance& response, AdminStream&) {
+  time_t current_time = time(nullptr);
```

Comment:
side note: we should probably be using TimeSystem here. cc @jmarantz. Don't need to action in this PR.

Reply:
Yes but I think that should be a separate refactor. I had not noticed all the calls to time(nullptr) -- I was looking for std::chrono.

This should stay consistent with the way time is captured in server/server.cc which is with time(nullptr).

Tracking with https://github.com/envoyproxy/envoy/issues/6953